homeassistant/components/sisyphus.py

```diff
@@ -0,0 +1,81 @@
+"""
+Enables control of Sisyphus Kinetic Art Tables.
+
+Each table is exposed as a light and a media player.
+"""
+import logging
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    EVENT_HOMEASSISTANT_STOP
+)
+from homeassistant.helpers.discovery import async_load_platform
+
+REQUIREMENTS = ['sisyphus-control==2.0.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SISYPHUS = 'sisyphus'
+DOMAIN = 'sisyphus'
+
+AUTODETECT_SCHEMA = vol.Schema({
+    DOMAIN: {},
+}, extra=vol.ALLOW_EXTRA)
+
+TABLES_SCHEMA = vol.Schema({
+    DOMAIN: [
+        {
+            vol.Required(CONF_NAME): cv.string,
+            vol.Required(CONF_HOST): cv.string,
+        },
+    ],
+}, extra=vol.ALLOW_EXTRA)
+
+CONFIG_SCHEMA = vol.Any(AUTODETECT_SCHEMA, TABLES_SCHEMA)
```

Comment:
Use something like

```python
CONFIG_SCHEMA = vol.Schema({
    DOMAIN: vol.Any(AUTODETECT_SCHEMA, TABLES_SCHEMA)
}, extra=vol.ALLOW_EXTRA)
```

This would make the two different parts shorter and easier to read.